core/cli/commands.py

```diff
@@ -47,17 +54,28 @@ def select_module_msg(module):
                         select_module_msg(self.module)
             else:
                 for k, v in self.modules.items():
-                        for name, module in v.items():
-                            if arg.lower() == name.lower():
-                                module_found = True
-                                self.module = name
-                                self.namespace = k
-                                select_module_msg(self.module)            
+                    for name, module in v.items():
+                        if arg.lower() == name.lower():
+                            module_found = True
+                            self.module = name
+                            self.namespace = k
+                            select_module_msg(self.module)
             if not module_found:
                 logger.error(f"No module/namespace named \"{arg}\".")
         else:
             logger.warn("'use' command takes 1 argument.")
 
+    def complete_use(self, text, line, begidx, endidx):
+        AVAILABLE_MODULES = self.modules.get(self.namespace)
+        if AVAILABLE_MODULES is None:  # if self.namespace == ''
+            AVAILABLE_MODULES = list()
+            for key, value in self.modules.items():
+                for name, module in value.items():
+                    AVAILABLE_MODULES.append(f"{key.upper()}.{name}")
```

Comment:
It is not finding the right module with this.
Should be: `AVAILABLE_MODULES.append(f"{key}/{name}")`

Reply:
And also I changed the function a little bit for the comfort of my fingers.
Maybe you will consider:
```
def complete_use(self, text, line, begidx, endidx):
        AVAILABLE_MODULES = self.modules.get(self.namespace)
        if AVAILABLE_MODULES is None:  # if self.namespace == ''
            AVAILABLE_MODULES = list()
            for key, value in self.modules.items():
                for name, module in value.items():
                    AVAILABLE_MODULES.append(f"{key}/{name}")
        mline = line.partition(' ')[2]
        offs = len(mline) - len(text)
        #######
        # Get completed text with namespace
        comp_text =  [s[offs:] for s in AVAILABLE_MODULES if s.startswith(mline)]
        # If no namespace found
        if len(comp_text) == 0:
            # Try to complete with module names
            AVAILABLE_MODULES = self.modules.get(list(self.modules.keys())[0])
            comp_text = [s[offs:] for s in AVAILABLE_MODULES if s.startswith(mline)]
        # Return
        return comp_text
        #######
```

Reply:
Another _nacizane_ idea of mine: Maybe we can move the part that responsible for creating the `AVAILABLE_MODULES` dict. It's just creating that dictionary over and over again on TAB key presses.

Reply:
+ I've updated the code above.
(Tried to solve case sensitivity issue.)